go/enclave/rpc/rpc_encryption_manager.go

```diff
@@ -95,3 +97,20 @@ func (rpc *EncryptionManager) EncryptWithViewingKey(address gethcommon.Address,
 
 	return encryptedBytes, nil
 }
+
+// AuthenticateSubscriptionRequest checks that a subscription request is authenticated correctly.
+func (rpc *EncryptionManager) AuthenticateSubscriptionRequest(subscription common.LogSubscription) error {
+	accountHashBytes := subscription.SubscriptionAccount.Account.Hash().Bytes()
+
+	recoveredViewingPublicKey, err := crypto.SigToPub(accountHashBytes, *subscription.SubscriptionAccount.Signature)
```

Comment:
does this also verify the signature over the: `accountHashBytes`

Reply:
Yeah, it does under the hood.